appsec-kit-starter/src/main/java/com/vaadin/appsec/service/AppSecDevToolsPlugin.java

```diff
@@ -17,23 +17,22 @@ public class AppSecDevToolsPlugin implements DevToolsMessageHandler {
     private static final Logger LOGGER = LoggerFactory
             .getLogger(AppSecDevToolsPlugin.class);
 
+    private boolean scanEventListenerAdded = false;
+
     @Override
     public void handleConnect(DevToolsInterface devToolsInterface) {
-        LOGGER.info("Plugin connected");
         devToolsInterface.send("appsec-kit-init", Json.createObject());
         AppSecService appSecService = AppSecService.getInstance();
-        appSecService.addScanEventListener(scanEvent -> {
-            LOGGER.info("Scan completed");
-            var vulnerabilityCount = scanEvent.getNewVulnerabilities().size();
-            var data = Json.createObject();
-            data.put("vulnerabilityCount", vulnerabilityCount);
-            devToolsInterface.send("appsec-kit-scan", data);
-            LOGGER.info("Vulnerabilities sent to the client: "
-                    + vulnerabilityCount);
-        });
-        LOGGER.debug("Scan event listener added for AppSec devtools plugin");
-        appSecService.scanForVulnerabilities()
-                .thenRun(appSecService::scheduleAutomaticScan);
+
+        if (!scanEventListenerAdded) {
+            appSecService.addScanEventListener(scanEvent -> sendScanResult(
+                    scanEvent.getNewVulnerabilities().size(),
+                    devToolsInterface));
+            scanEventListenerAdded = true;
+            LOGGER.debug("Scan event listener added");
+        }
+        refreshScanResult(appSecService.getNewVulnerabilities().size(),
+                devToolsInterface);
     }
```

Comment:
This unfortunately can't work: while the `AppSecDevtoolsPlugin` instance is the same, the `handleConnect` might receive different `DevToolsInterface` objects (referring to different browsers) and we need to add the listener for each of them.

One possible solution could be to save a `<DevToolsInterface, Registration>` map, so we can also unregister the listeners when the plugin disconnects (or re-connects), e.g.

```suggestion
        if (!scanEventRegistrations.containsKey(devToolsInterface)) {
            Registration scanEventRegistration = appSecService
                    .addScanEventListener(event -> {
                        sendData("appsec-kit-scan",
                                event.getNewVulnerabilities().size(),
                                devToolsInterface);
                    });
            scanEventRegistrations.put(devToolsInterface,
                    scanEventRegistration);
        }

        sendData("appsec-kit-refresh",
                appSecService.getNewVulnerabilities().size(),
                devToolsInterface);
```

Reply:
Good point. In this case I will use the suggested solution.

Reply:
Tried this solution and on each page refresh, new window and new tab a new `DevToolsInterface` object is created and added to the `scanEventRegistrations`. There is no possibility to remove the old ones because there is no plugin disconnected (or re-connected) mechanism to use (there are no events about these).

Reply:
I [committed](https://github.com/vaadin/appsec-kit/pull/113/commits/ded79f450941f63df5239a4fe0d549854ed3dcba) this solution for testing purpose.